js/components/portal-dashboard/all-responses-popup/student-responses-popup.tsx

```diff
@@ -103,9 +109,23 @@ export class StudentResponsePopup extends React.PureComponent<IProps, IState> {
     );
   }
 
+  private handleChangeColorGroup = (studentId: string) => {
+    const { selectedStudents } = this.state;
+    const index = selectedStudents.findIndex((s: SelectedStudent) => s.id === studentId);
+    if (index >= 0) {
+      const currentColor = selectedStudents[index].colorGroup;
+      const newColor = currentColor >= spotlightColors.length - 1 ? 0 : currentColor + 1;
+      const updatedStudent: SelectedStudent = { id: studentId, colorGroup: newColor };
+      const updatedSelectedStudents = [...selectedStudents];
+      updatedSelectedStudents.splice(index, 1);
+      updatedSelectedStudents.push(updatedStudent);
```

Comment:
As long as we're dealing with simple JavaScript objects (e.g. not immutable objects), you don't actually need to splice in a new student object. Another approach in this case, by the way, would be a map which passed most objects through unchanged and only changed the one of interest, e.g. something like:
```
  const updatedSelectedStudents = selectedStudents.map(s => {
    return (s.id === studentId) ? { id: studentId, colorGroup: newColor } : s;
  });